File: Modules/SEDO/EMS/SIE.Web.EMS/SpareParts/Commands/AddSparePartStoreCommand.cs
using SIE.EMS.SpareParts;
using SIE.Web.Command;

namespace SIE.Web.EMS.SpareParts.Commands
{
    /// <summary>
    /// 备件入库添加命令
    /// </summary>
    [JsCommand("SIE.Web.EMS.SpareParts.Commands.AddSparePartStoreCommand")]
    public class AddSparePartStoreCommand : ViewCommand
    {
        /// <summary>
        /// 执行添加操作
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>工作任务号</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var No = RT.Service.Resolve<SparePartController>().GetStoreCode();
            return No;
        }
    }
}

File: Modules/SEDO/EMS/SIE.Web.EMS/SpareParts/Commands/AddSparePartStoreCommand.js
SIE.defineCommand('SIE.Web.EMS.SpareParts.Commands.AddSparePartStoreCommand', {
    extend: 'SIE.cmd.Add',
    meta: { text: "添加", group: "edit" }, 
    onItemCreated: function (entity) {
        if (entity) {
            var model = entity.data;
            var me = this;
            this.view.execute({
                data: [],
                success: function (res) {
                    entity.setStoreCode(res.Result);
                }
            }, me.view);
        }
    },
});